### ISSUE TITLE
Failure to correctly unpack paths for roads that allow u-turns at both ends

### ISSUE THREAD

**Issue by MoKob** [Project-OSRM/osrm-backend#3359]
Right now we mostly forbid u-turns. In cases where they are explicitly allowed, we fail to unpack paths correctly.

The following scenario:

```
     Scenario: U-Turn In Bearings
        Given the node map
            """
            a 1 b
            """

        And the ways
            | nodes |
            | ab    |

        When I route I should get
            | waypoints | bearings   | route    | turns                    |
            | 1,a       | 90,2 270,2 | ab,ab,ab | depart,turn uturn,arrive |
            | 1,b       | 270,2 90,2 | ab,ab,ab | depart,turn uturn,arrive |
```

Fails to succeed currently.

Based on which direction the shortcut is first added, only one of the two paths will be found correctly.

Right now we see

```
1) Scenario: U-Turn In Bearings - features/testbot/via.feature:331
   Step: When I route I should get - features/testbot/via.feature:341
   Step Definition: features/step_definitions/routing.js:4
   Message:
     Error: Tables were not identical:
         waypoints |     bearings |     route |     turns
     |     1,a |     90,2 270,2 |     ab,ab,ab |     depart,turn uturn,arrive | 
     | (-) 1,b | (-) 270,2 90,2 | (-) ab,ab,ab | (-) depart,turn uturn,arrive | 
     | (+) 1,b | (+) 270,2 90,2 | (+) ab,ab | (+) depart,arrive | 
```

**Comment by MoKob** [comment]
A test case has been added [here](https://github.com/Project-OSRM/osrm-backend/blob/3721f8c8872b17f0b1d00f8ccffe78dfbca21c94/features/testbot/via.feature#L331-L345). Issue up for the taking.

**Comment by danpat** [comment]
This problem was fixed at some point in the last 12 months.  I removed the `@todo` tag in https://github.com/Project-OSRM/osrm-backend/pull/4712, as the tests now pass.

**Comment by oxidase** [comment]
@danpat the issues was fixed by 40d02978858502ed5f08fa7aa69bb9afc72b2515 the problem was in path unpacking.
